api/v1alpha1/jobset_webhook.go

```diff
@@ -56,6 +57,14 @@ var _ webhook.Validator = &JobSet{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *JobSet) ValidateCreate() error {
+	for i := range r.Spec.ReplicatedJobs {
+		if r.Spec.ReplicatedJobs[i].Network == nil || !*r.Spec.ReplicatedJobs[i].Network.EnableDNSHostnames ||
+			(r.Spec.ReplicatedJobs[i].Template.Spec.CompletionMode != nil &&
+				*r.Spec.ReplicatedJobs[i].Template.Spec.CompletionMode == batchv1.IndexedCompletion) {
+			// how do we record the error message here?
+			fmt.Println("EnableDNSHostnames requires job to be in indexed completion mode")
```

Comment:
I think you can just do `return  fmt.Errorf("EnableDNSHostnames requires job to be in indexed completion mode")`